Packs/Kafka/Integrations/KafkaV3/KafkaV3.py

```diff
@@ -60,55 +59,93 @@ def __init__(self, brokers: str, offset: str = 'earliest', group_id: str = 'xsoa
             self.conf_consumer.update({'message.max.bytes': int(message_max_bytes)})
 
         demisto.debug(f"The consumer configuration is \n{self.conf_consumer}\n")
         demisto.debug(f"The producer configuration is \n{self.conf_producer}\n")
 
-        if ca_cert:
-            with tempfile.NamedTemporaryFile(mode="w", delete=False) as ca_descriptor:
-                self.ca_path = ca_descriptor.name
-                ca_descriptor.write(ca_cert)
-            self.conf_producer.update({'ssl.ca.location': self.ca_path})
-            self.conf_consumer.update({'ssl.ca.location': self.ca_path})
-        if client_cert:
-            with tempfile.NamedTemporaryFile(mode="w", delete=False) as client_cert_descriptor:
-                self.client_cert_path = client_cert_descriptor.name
-                client_cert_descriptor.write(client_cert)
-            self.conf_producer.update({'ssl.certificate.location': self.client_cert_path,
-                                       'security.protocol': 'ssl'})
-            self.conf_consumer.update({'ssl.certificate.location': self.client_cert_path,
-                                       'security.protocol': 'ssl'})
-        if client_cert_key:
-            with tempfile.NamedTemporaryFile(mode="w", delete=False) as client_key_descriptor:
-                self.client_key_path = client_key_descriptor.name
-                client_key_descriptor.write(client_cert_key)
-            self.conf_producer.update({'ssl.key.location': self.client_key_path})
-            self.conf_consumer.update({'ssl.key.location': self.client_key_path})
-        if ssl_password:
-            self.conf_producer.update({'ssl.key.password': ssl_password})
-            self.conf_consumer.update({'ssl.key.password': ssl_password})
+    def update_client_dict(self, client_dict, trust_any_cert, use_ssl, ca_cert, client_cert, client_cert_key, ssl_password,
+                           use_sasl, plain_username, plain_password, brokers):
+        """
+        Updates the conf_producer or conf_consumer configuration based on the specified authentication method.
+        It assumes that all required parameters have been validated by the validate_params function.
+
+        Args:
+            client_dict (dict): The configuration dictionary to be updated.
+            This should be either the producer's or consumer's conf_dict.
+        """
+
+        client_dict.update({'bootstrap.servers': brokers})
+
+        if use_ssl and not use_sasl:
+            if self.ca_path:
+                client_dict.update({'ssl.ca.location': self.ca_path, 'ssl.certificate.location': self.client_cert_path,
+                                    'ssl.key.location': self.client_key_path, 'security.protocol': 'ssl'})
+            else:
+                # ca_cert
+                with tempfile.NamedTemporaryFile(mode="w", delete=False) as ca_descriptor:
+                    self.ca_path = ca_descriptor.name
+                    ca_descriptor.write(ca_cert)
```

Comment:
We can take that to an helper function as it happens 3 times.

Reply:
I considered that approach initially, but I found that it had a negative impact on the code's readability.